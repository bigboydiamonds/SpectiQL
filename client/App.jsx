import React, { Component } from "react";
import { HashRouter, Route, Link, Switch } from "react-router-dom";
import Main from "./mainContainer/main.jsx";

import './public/styling/index.css';
// const createSchema = require('../spectiql').createSchema
// import { createSchema } from '../spectiql'


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      text: "sodium bloat",
      filePath: '',
      schema: [],
    };
  }

  openDocs() {
    window.open(
      "https://github.com/oslabs-beta/SpectiQL/blob/master/README.md"
    );
  }



        render() {
            return (
                <HashRouter> 
                    <div className="fullscreen">
                    <div className="input-group">
                        <span className="input-group-text" id="inputGroupFileAddon01">
                          Upload
                        </span>
                      <div className="custom-file">
                        <input
                          type="file"
                          className="custom-file-input"
                          id="inputGroupFile01"
                          aria-describedby="inputGroupFileAddon01"
                          onChange={(e) => {this.setState({filePath: e.target.files[0].name})}}
                        />
                        <label className="custom-file-label" htmlFor="inputGroupFile01">
                          {this.state.filePath}
                        </label>
                        </div>
                      </div>
                   <div className="introContainer">
                      <div className="introHeader">
                        SpectiQL
                      </div>
                                    <div className="introNext">
                                    <Link to="/main" exact>
                                        <button className="next-button">Next</button>
                                        </Link>
                                    </div>
                                    <div className="introDoc">
                                    <Link to="/documentation" exact onClick={this.openDocs}>
                                        <button className="doc-button">Docs</button>
                                        </Link>
                                        </div>
                        <Switch>
                            <Route path="/main" exact component={Main} />
                        </Switch>
                    </div>
                    </div>
                </HashRouter>
            )
        }
};

export default App;
