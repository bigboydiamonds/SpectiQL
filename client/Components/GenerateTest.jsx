import React, { Component } from 'react'
import { Button, Form, Col } from 'react-bootstrap'

class GenerateTest extends Component {
  constructor (props) {
    super(props)
  };

  render () {
    const editTestSuite = () => {
      if (this.props.testSuiteToggler === true) {
        return (
          <Button
            className='addTestButton'
            variant='outline-primary'
            size='lg'
            block
            style={{ height: '42%', width: '100%', color: 'black', borderColor: 'black' }}
            onClick={this.props.addTestSuite}
          >
          Add to Test Suite
          </Button>
        )
      } else {
        return (
          <Button
            className='updateTest'
            variant='outline-primary'
            size='lg'
            block
            style={{ height: '42%', width: '100%', color: 'black', borderColor: 'black' }}
            onClick={this.props.updateTestSuite}
          >
          Update Test Suite
          </Button>
        )
      }
    }
    return (
      <div className='generateTestBox'>
        {/* <Form className="generate-test"> */}
        {/* <Form.Row> */}
        <Form.Group as={Col} style={{ margin: 0, padding: 3 }}>
          <Button
            className='generateTestButton'
            variant='outline-primary'
            size='lg'
            block
            style={{ height: '42%', width: '100%', color: 'black', borderColor: 'black' }}
            onClick={this.props.handleClick}
          >
              Generate Test
          </Button>
          {/* <Button className="addTestButton"
              variant="outline-primary"
              size="lg"
              block
              style={{ height: "42%", width: "100%", color: "black", borderColor: "black" }}
              onClick={this.props.updateTestSuite}
            >
              Add to Test Suites
            </Button> */}
          {editTestSuite()}
        </Form.Group>

        <Form.Group as={Col} controlId='sample-test'>
          {/* <Form.Label column sm={6}>
              Generated Test:
            </Form.Label> */}
          <Form.Control
            as='textarea'
            placeholder='Generated Test...'
            style={{ width: '100%', height: '17.7rem' }}
            value={this.props.generatedTest}
          />
        </Form.Group>
        {/* </Form.Row> */}
        {/* </Form> */}
      </div>
    )
  }
}

export default GenerateTest
