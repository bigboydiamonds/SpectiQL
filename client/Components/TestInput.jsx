import React, { Component } from "react";
import { Form, Col } from "react-bootstrap";

class TestInput extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="testInputContainer">
        {/* <Form className="testSuiteInput"> */}
          <Form.Group controlId="test-suite">
            {/* <Form.Label column sm={6}>
              Test Suite:
            </Form.Label> */}
            <Col sm="10">
              <Form.Control
                type="text"
                placeholder="Enter test suite here..."
                name="testSuiteName"
                value={this.props.testSuiteName}
                onChange={this.props.handleChange}
              />
            </Col>
          </Form.Group>
        {/* </Form> */}
            
        {/* <Form className="testDescriptionInput"> */}
          <Form.Group controlId="test-description">
            {/* <Form.Label column sm={6}>
              Test Description:
            </Form.Label> */}
            <Col sm="10">
              <Form.Control
                type="text"
                placeholder="Enter test description here..."
                name="testDescription"
                value={this.props.testDescription}
                onChange={this.props.handleChange}
              />
            </Col>
          </Form.Group>
        {/* </Form> */}
        <select className="selectQueryType" id="dd-reset" onChange={this.props.selectTest}>
              <option  value ="default" disabled selected>Select Test</option>
              <option  value = "validQuery">Valid Query</option>
              <option  value= "invalidQuery" >Invalid Query</option>
              <option  value= "validArgField" >Valid Argument Field</option>
              <option  value = "invalidArgField" >Invalid Argument Field</option>
              <option  value= "validArgDataType" >Valid Argument DataType</option>
              <option  value = "invalidArgDataType" >Invalid Argument DataType</option>
            </select>
      </div>
    );
  }
}
  
export default TestInput;
