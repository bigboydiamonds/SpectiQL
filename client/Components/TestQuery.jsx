import React, { Component } from "react";
import { Form, Col } from "react-bootstrap";

class TestQuery extends Component {
  constructor(props) {
    super(props);
  };

  render() {
    // const schemaTreeData = [
    //   {
    //     name: 'Query',
    //     attributes: {
    //       type: 'Root'
    //     },
    //   },
    //   children:[ {}]
    // ];
    console.log(this.props.schema.queries);
    
    return (
      <div className="writeQueryBox">
      {/* <Form className="test-query-box"> */}
        {/* <Form.Row> */}
          <div className= "test-query-write">
          <Form.Group as={Col} controlId="write-query">
            {/* <Form.Label column sm={6}>
              Write Query:
            </Form.Label> */}
            <Form.Control
              as="textarea"
              placeholder="Enter query here..."
              style={{ width: "100%", height: "17.7rem" }}
              name = "writeQuery"
              value = {this.props.writeQuery}
              onChange = {this.props.onChange}
            />
          </Form.Group>
          </div>
          
          <div className= "test-query-select">
          <Form.Group as={Col} controlId="select-query">
            {/* <Form.Label column sm={6}>
              Select Query:
            </Form.Label> */}
            <Form.Control
              as="textarea"
              placeholder="Sample selected query here..."
              style={{ width: "100%", height: "17.7rem" }}
            />
          </Form.Group>
          </div>
        {/* </Form.Row> */}
      {/* </Form> */}
      </div>
    );
  }
}

export default TestQuery;
