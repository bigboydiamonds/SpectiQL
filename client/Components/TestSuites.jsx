import React, { Component } from "react";
import { Card, Button } from "react-bootstrap";

class TestSuites extends Component {
  render() {

    const testSuite = this.props.testSuites.map((test, index) => (
      // console.log(`test`, test)
      <Card style={{  }} key={index}>
        <Card.Body>
          <Card.Title>Test Suite {index + 1}</Card.Title>
          <Card.Subtitle className="mb-2 text-muted">
            {test.savedTestSuiteName}
          </Card.Subtitle>
          <Card.Text>{test.savedTestDescription}</Card.Text>
          <button className="testSuiteEditButton"
            onClick={() => this.props.editTest(test.testIndex)}
          >
            Edit Test
          </button>
          <button className="testSuiteDeleteButton"
            onClick={() => this.props.deleteTest(test.testIndex)}
          >
            Delete Test
          </button>
        </Card.Body>
      </Card>
    ));
    return (
      <div class="wrapper">
        <h3>Test Suites</h3>
        <div id="sidebar">
          <div class="sidebar-header">
            {/* <h3>Test Suites</h3> */}
          </div>
          {testSuite}
        </div>

        <div id="content">
          <nav id="content" class="navbar navbar-expand-lg navbar-light bg-light">
            {/* <div class="container-fluid">
              <button type="button" id="sidebarCollapse" class="btn btn-info">
                <i class="fas fa-align-left"></i>
                <span>Toggle Sidebar</span>
              </button>
            </div> */}
          </nav>
        </div>
      </div>
    );
  }
}

export default TestSuites;
