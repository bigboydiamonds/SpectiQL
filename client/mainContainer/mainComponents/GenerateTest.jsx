import React, { Component } from "react";
import { Button, Form, Col, DropdownButton, Dropdown } from "react-bootstrap";

class GenerateTest extends Component {
  render() {
    return (
      <Form className="generate-test">
        <Form.Row>
          <Form.Group as={Col} style={{ margin: 0, padding: 0 }}>
            <select  onChange = {this.props.selectTest}>
              <option value = "" selected disabled>Select Test</option>
              <option value = "validQuery">Valid Query</option>
              <option value= "invalidQuery" >Invalid Query</option>
              <option value= "validArgField" >Valid Argument Field</option>
              <option value = "invalidArgField" >Invalid Argument Field</option>
              <option value= "validArgDataType" >Valid Argument DataType</option>
              <option value = "invalidArgDataType" >Invalid Argument DataType</option>
            </select>
            <Button
              variant="primary"
              size="lg"
              block
              style={{ height: "4rem", width: "80%" }}
              onClick={this.props.handleClick}
            >
              Generate Test
            </Button>
            <Button
              variant="outline-primary"
              size="lg"
              block
              style={{ height: "4rem", width: "80%" }}
              onClick={this.props.updateTestSuite}
            >
              Add to Test Suite
            </Button>
          </Form.Group>

          <Form.Group as={Col} controlId="sample-test">
            <Form.Label column sm={3}>
              Generated Test:
            </Form.Label>
            <Form.Control
              as="textarea"
              placeholder="Sample Test..."
              style={{ width: "80%", height: "18rem" }}
              value={this.props.generatedTest}
            />
          </Form.Group>
        </Form.Row>
      </Form>
    );
  }
}

export default GenerateTest;
