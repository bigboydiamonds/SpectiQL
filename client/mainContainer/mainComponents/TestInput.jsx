import React, { Component } from "react";
import { Form, Col } from "react-bootstrap";

class TestInput extends Component {
  render() {
    return (
      <div className="testInputContainer">
        {/* <Form className="testSuiteInput"> */}
          <Form.Group controlId="test-suite">
            {/* <Form.Label column sm={6}>
              Test Suite:
            </Form.Label> */}
            <Col sm="10">
              <Form.Control
                type="text"
                placeholder="Enter test suite here..."
                name="testSuiteName"
                value={this.props.testSuiteName}
                onChange={this.props.onChange}
              />
            </Col>
          </Form.Group>
        {/* </Form> */}
            
        {/* <Form className="testDescriptionInput"> */}
          <Form.Group controlId="test-description">
            {/* <Form.Label column sm={6}>
              Test Description:
            </Form.Label> */}
            <Col sm="10">
              <Form.Control
                type="text"
                placeholder="Enter test description here..."
                name="testDescription"
                value={this.props.testDescription}
                onChange={this.props.onChange}
              />
            </Col>
          </Form.Group>
        {/* </Form> */}
        <select className="selectQueryType" id="dd-reset" onChange = {this.props.selectTest}>
              <option className="new-select" value ="default" disabled selected>Select Test</option>
              <option className="new-select" value = "validQuery">Valid Query</option>
              <option className="new-select" value= "invalidQuery" >Invalid Query</option>
              <option className="new-select" value= "validArgField" >Valid Argument Field</option>
              <option className="new-select" value = "invalidArgField" >Invalid Argument Field</option>
              <option className="new-select" value= "validArgDataType" >Valid Argument DataType</option>
              <option className="new-select" value = "invalidArgDataType" >Invalid Argument DataType</option>
            </select>
      </div>
    );
  }
}

export default TestInput;
