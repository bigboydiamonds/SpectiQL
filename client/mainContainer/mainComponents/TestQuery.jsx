import React, { Component } from "react";
import { Form, Col } from "react-bootstrap";

class TestQuery extends Component {
  render() {
    return (
      <Form className="test-query-box">
        <Form.Row>
          <Form.Group as={Col} controlId="write-query">
            <Form.Label column sm={3}>
              Write Query:
            </Form.Label>
            <Form.Control
              as="textarea"
              placeholder="Enter query here..."
              style={{ width: "80%", height: "18rem" }}
              name = "writeQuery"
              value = {this.props.writeQuery}
              onChange = {this.props.onChange}
            />
          </Form.Group>

          <Form.Group as={Col} controlId="select-query">
            <Form.Label column sm={3}>
              Select Query:
            </Form.Label>
            <Form.Control
              as="textarea"
              placeholder="Sample selected query here..."
              style={{ width: "80%", height: "18rem" }}
            />
          </Form.Group>
        </Form.Row>
      </Form>
    );
  }
}

export default TestQuery;
