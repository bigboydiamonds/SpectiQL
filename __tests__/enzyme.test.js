import App from '../client/App.jsx';
import GenerateTest from '../client/Components/GenerateTest.jsx';
import TestInput from '../client/Components/TestInput.jsx';
import TestSuites from '../client/Components/TestSuites.jsx';
import TestQuery from '../client/Components/TestQuery.jsx';
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

// testing App component
describe('testing App component', () => {
   let wrapper;
   beforeAll(() => { wrapper = shallow(<App />); });

   it('renders without crashing', () => {
      expect(wrapper.length).toEqual(1);
   });

  //  it('links back to the intro page', () => {
  //     expect(wrapper.find('Link').props('to')).toEqual('/main')
  //  });
});

// testing Generate Test for Queries
describe('testing Generate Test component', () => {
   let wrapper;
   const props = {
    testDescription: 'test description',
    onClick: jest.fn(),
    testSuiteName: 'test suite name',
    value: 'random value'
   }
   beforeAll(() => { wrapper = shallow(<GenerateTest {...props}/>) });

   it('contains a form with 2 buttons', () => {
      expect(wrapper.find('Button')).toHaveLength(2);
   });

  //  it('functions should be invoked on the onClick', () => {
  //    const buttonArray = wrapper.find('Button'); 
  //    buttonArray.forEach(button => {
  //     button.simulate('click')
  //    })
  //    expect(props.onClick).toHaveBeenCalled();
  //  })
  

   //onclick button should work
  //  it('onClick should trigger', () => {
  //   wrapper.find('Button').simulate('click');
  //   expect(mockCallBack.mock.calls.length).toEqual(1);
  //  })

});

// testing for TestInput
describe('testing Test Input component', () => {
   let wrapper;
  //  const props = {
     
  //  }
   beforeAll(() => { wrapper = shallow(<TestInput/>) })
//    it('render correctly date component', () => {  
//     const DateInputComponent = renderer.create(<DateInput />).toJSON();
//     expect(DateInputComponent).toMatchSnapshot();
// });
   // need to test this
   it('snapshot testing', () => {
      expect(wrapper).toMatchSnapshot();
   })

    it('has 7 options within dropdown', () => {
      expect(wrapper.find('option')).toHaveLength(7);
   })
   // iffy about this test may need refactoring
  //  it('simulates a mock onchange value to test functionality', () => {
  //     const text = wrapper.find('Form.Control');

  //     // expect(text.at(0).prop('value')).toEqual('test suite name');
  //     // expect(text.at(1).prop('value')).toEqual('test description');
  //  })
})

// test for Test Suites
describe('testing Test Suite component', () => {
   let wrapper;
   const props = {
      editTest: jest.fn(),
      deleteTest: jest.fn()
   }
   beforeAll(() => { wrapper = shallow(<TestSuites {...props} />)})

   it('should fire onclick when button is pressed', () => {
      const buttonArray = wrapper.find('button');
      buttonArray.forEach(button => button.simulate('click'))
      expect(props.editTest).toHaveBeenCalled();
      expect(props.deleteTest).toHaveBeenCalled();
   });

   it('should have H3 element to display Test Suites', () => {
      let h3 = wrapper.find('h3');
      expect(h3.text()).toBe('Test Suites');
   });
})

// test for TestQuery
describe('testing TestQuery component', () => {
   let wrapper;
   beforeAll(() => { wrapper = shallow(<TestQuery />)});
   
   it('should render 2 Form Groups for write query and select query', () => {
      expect(wrapper.find('div.test-query-write')).toHaveLength(1);
      expect(wrapper.find('div.test-query-select')).toHaveLength(1);
   })
})