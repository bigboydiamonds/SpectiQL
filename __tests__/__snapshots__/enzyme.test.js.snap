// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Generate Test component Snapshot testing 1`] = `
<div
  className="generateTestBox"
>
  <FormGroup
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Col",
        "render": [Function],
      }
    }
    style={
      Object {
        "margin": 0,
        "padding": 3,
      }
    }
  >
    <Button
      active={false}
      block={true}
      className="generateTestButton"
      disabled={false}
      size="lg"
      style={
        Object {
          "borderColor": "black",
          "color": "black",
          "height": "42%",
          "width": "100%",
        }
      }
      type="button"
      variant="outline-primary"
    >
      Generate Test
    </Button>
    <Button
      active={false}
      block={true}
      className="updateTest"
      disabled={false}
      size="lg"
      style={
        Object {
          "borderColor": "black",
          "color": "black",
          "height": "42%",
          "width": "100%",
        }
      }
      type="button"
      variant="outline-primary"
    >
      Update Test Suite
    </Button>
  </FormGroup>
  <FormGroup
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Col",
        "render": [Function],
      }
    }
    controlId="sample-test"
  >
    <FormControl
      as="textarea"
      placeholder="Generated Test..."
      style={
        Object {
          "height": "17.7rem",
          "width": "100%",
        }
      }
    />
  </FormGroup>
</div>
`;

exports[`testing Test Input component snapshot testing 1`] = `ShallowWrapper {}`;

exports[`testing TestQuery component Snapshot testing 1`] = `
<div
  className="writeQueryBox"
>
  <div
    className="test-query-write"
  >
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Col",
          "render": [Function],
        }
      }
      controlId="write-query"
    >
      <FormLabel
        column={true}
        sm={6}
        srOnly={false}
      >
        Write Query:
      </FormLabel>
      <FormControl
        as="textarea"
        name="writeTest"
        placeholder="Enter query here..."
        style={
          Object {
            "height": "17.7rem",
            "width": "100%",
          }
        }
      />
    </FormGroup>
  </div>
</div>
`;

exports[`testing for leftSideBar Snapshot testing 1`] = `
<div
  className="mainNavBar"
>
  <ButtonToolbar
    className="buttonContainer"
    role="toolbar"
  >
    <Link
      to="/queries"
    >
      <Button
        active={false}
        className="queries"
        disabled={false}
        size="lg"
        type="button"
        variant="outline-dark"
      >
        Queries
      </Button>
    </Link>
    <Link
      to="/mutations"
    >
      <Button
        active={false}
        className="mutations"
        disabled={false}
        size="lg"
        type="button"
        variant="outline-dark"
      >
        Mutations
      </Button>
    </Link>
    <Button
      active={false}
      className="subscriptions"
      disabled={false}
      size="lg"
      type="button"
      variant="outline-dark"
    >
      Subscriptions
    </Button>
    <Button
      active={false}
      className="export"
      disabled={false}
      size="lg"
      type="button"
      variant="outline-dark"
    >
      Export
    </Button>
  </ButtonToolbar>
</div>
`;
